### PR TITLE
Add a `nonstd::camel_case` util function

### PR DESCRIPTION
**Pull request by eddyashton**
Useful for `https://github.com/microsoft/CCF/pull/6326#issuecomment-2214510569`, potentially other situations.

### CODE REVIEW

File: include/ccf/ds/nonstd.h [hunk 1]
```diff
@@ -288,4 +292,32 @@ namespace ccf::nonstd
 
     return resolved.lexically_normal().string();
   }
+
+  static inline std::string camel_case(
```
Comment (achamayou) [inline include/ccf/ds/nonstd.h:296]: Let's start a `src/ds/nonstd.cpp` and stop inlining everything.

**Review by achamayou — APPROVED** [review]

**Comment by lemmy** [comment]
FYI: Trying this PR out in the context of #6326 right now.

**Comment by lemmy** [comment]
@eddyashton Would camel-casing all-caps strings such as "POST", "DELETE", ... also be an input that the function could/should accept?

File: include/ccf/ds/nonstd.h (hunk 1 above)
Comment (achamayou) [reply include/ccf/ds/nonstd.h:296]: Actually let's not export that in public headers either.

**Comment by eddyashton** [comment]
Flattened this into #6326, will add followup (making definition private) next.

**Pull request closed without merging**